packages/core/src/utils/parseInvitation.ts

```diff
@@ -102,9 +105,33 @@ export const parseInvitationShortUrl = async (
   if (parsedUrl['oob']) {
     const outOfBandInvitation = OutOfBandInvitation.fromUrl(invitationUrl)
     return outOfBandInvitation
-  } else if (parsedUrl['c_i'] || parsedUrl['d_m']) {
+  } else if (parsedUrl['c_i']) {
     const invitation = ConnectionInvitationMessage.fromUrl(invitationUrl)
     return convertToNewInvitation(invitation)
+  }
+  // Legacy connectionless invitation
+  else if (parsedUrl['d_m']) {
+    const messageJson = JsonEncoder.fromBase64(parsedUrl['d_m'] as string)
+    const agentMessage = JsonTransformer.fromJSON(messageJson, AgentMessage)
+
+    // ~service is required for legacy connectionless invitations
+    if (!agentMessage.service) {
+      throw new AriesFrameworkError('Invalid legacy connectionless invitation url. Missing ~service decorator.')
+    }
+
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    const { '~service': service, ...messageWithoutService } = messageJson
+
+    // transform into out of band invitation
+    const invitation = new OutOfBandInvitation({
```

Comment:
If we can have out of band invitations without a label it should be moved to optional in the 0.5.0 as it would be breaking, right?

Maybe it can also be a param provided by the user as they sometimes might know the label when scanning?

Reply:
Out of band invitations do require a label I think. It's just that legacy connectionless invitations do not have one. Do you think we should make it optional for that? I see it as a legacy behaviour where we convert it to the new api (which has different requirements)

Reply:
Hmm so it seems the label is optional. So I think we should make it optinoal in AFJ as well, but that would be a breaking change. I'm going to leave it for now as an empty string if that's okay (or a default value), and before 0.5.0 we can make the property optional.

It would be a really simple change. Opened https://github.com/hyperledger/aries-framework-javascript/issues/1524